simpleviewinc/tap-events-force
# Update Button Children Styles

**Ticket**: [ZEN-581](https://jira.simpleviewtools.com/browse/ZEN-581)

> This PR should be tested in conjunction with [this keg-hub PR](https://github.com/simpleviewinc/keg-hub/pull/155)

## Context
* The UK team need more adjustments in order for them to control the styling fully
* They need to target elements more easily and override colors without using !important

## Goal
 * Don't apply very specific colour styles on button text, counter and tick svg so that we can override the colours by targeting ef classes on them
 * Add an ef class to button text element "ef-button-text"
 * Add an ef class to counter element "ef-button-text-counter"
 * Add an ef class to tick element "ef-button-text-icon"
 * Remove border colour from the tick element to match the counter element
 * "Clear all" in filter modal should use the ef passed in button with buttonType = "link"


### Updates
* `package.json`
  * Updated keg-core and jsutils versions
  * Updated to force jsutils to latest version
* `container/mutagen.yml`
  * Added install action in mutagen config
* `src/components/button/bookingButton.js`
  * Added classNames to EVF Button children based on ticket request
* `src/components/button/evfButton.js` && `src/theme/components/button/evfButton.js`
  * Removed all color from EVF Button styles
* `src/components/modals/filter.js`
  * Updated filter model clear all button to be a Link Button
  * Updated to use EVFButton
* `src/mocks/eventsforce/evfButton.js`
  * Added new link button type
  * Fixed render bug with LabelButton


### Tests

**Setup**
* Run this package => `keg evf pack run evf:zen-581-button-text-updates`
* Navigate to http://evf-zen-581-button-text-updates.local.keghub.io in the browser

**Test 1**
* Open your browsers inspector and inspect any button
  * Ensure that the child test elements have `ef-button-text` class applied to them
  * [Looks like this](https://prnt.sc/12862ff)

**Test 2**
* Register for any session, so that is shows a number on the button
  * Inspect that button, and ensure the counter has the `ef-button-text-counter` class applied to it
  * [Looks like this](https://prnt.sc/12862ff)

**Test 3**
* Using the `Test Data state switcher`, switch to a `single` mode, and Apply Changes
  * [Looks like this](https://prnt.sc/12863kx)
* Select for any session, so that is shows a check-mark on the button
  * Inspect that button, and ensure the counter has the `ef-button-text-icon` class applied to it
  * Ensure no border is applied to the icon
  * [Looks like this](https://prnt.sc/12874bs)

**Test 4**
* Open the filter modal and select one of the items
* Ensure the `Clear all` button appears, and ensure it's using the `link` button type
* [Looks like this](https://prnt.sc/12876ob)
